src/main/java/gov/nasa/cumulus/CnmToGranuleHandler.java

```diff
@@ -180,6 +187,17 @@ public String PerformFunction(String input, Context context) throws Exception {
 			}
 			granuleFile.addProperty("type", cnmFile.get("type").getAsString());
 
+
+			// If this translator receiving input as CNMResponse "style".  That is, this translator is being used
+			// to translate CNM Response message to granules.
+			if (isCNMResponseFileObject && ObjectUtils.isNotEmpty(cnmFile.get("uri"))) {
+				ResponseUriDecoder decoder = new ResponseUriDecoder();
+				ExtraFileFileds extra = decoder.process(cnmFile.get("uri").getAsString(), distribution_endpoint);
+				granuleFile.addProperty("bucket", extra.getBucket());
+				granuleFile.addProperty("filename", extra.getFilename());
```

Comment:
Is filename and filepath needed?

Reply:
I think filename is needed because the input of HyraxProcessing needs filename in there.
 "task_config": {
              "buckets": "{$.meta.buckets}",
              "distribution_endpoint": "{$.meta.distribution_endpoint}",
              "files_config": "{$.meta.collection.files}",
              "fileStagingDir": "{$.meta.collection.name}",
              "granuleIdExtraction": "{$.meta.collection.granuleIdExtraction}",
              "collection": "{$.meta.collection}",
              "cumulus_message": {
                "input": "{[$.payload.granules[*].files[*].filename]}"
              }
Not sure about filepath.  I will take some test to prove that.